src/app/components/week-selector/week-selector.component.ts

```diff
@@ -0,0 +1,69 @@
+import { async } from '@angular/core/testing';
+import { Component, OnInit, Input, Output, EventEmitter, AfterContentChecked } from '@angular/core';
+import { addWeeks, subWeeks, subYears, getWeek, getYear } from 'date-fns';
+
+@Component({
+  selector: 'app-week-selector',
+  templateUrl: './week-selector.component.html',
+  styleUrls: ['./week-selector.component.scss']
+})
+export class WeekSelectorComponent implements OnInit {
+
+  @Input() currentDate: Date;
+  @Input() startDate: Date;
+  @Input() subTitle: string;
+
+  @Output() passDate = new EventEmitter<number>();
```

Comment:
Ik vind het altijd een goede practice om deze uit te schrijven, aangezien je editor niet oppikt wat dit doet (VSCode in ieder geval niet). Als je dus besluit dat `startDate` eigenlijk `initDate` ofzo moet heten en deze hernoemt, werkt je component @Input niet meer.

Dus `@Input('startDate') startDate: string`;